python/global_scheduler/test/test.py

```diff
@@ -43,47 +46,69 @@ def random_task_id():
 def random_function_id():
   return photon.ObjectID(np.random.bytes(ID_SIZE))
 
+def random_object_id():
+  return photon.ObjectID(np.random.bytes(ID_SIZE))
+
 def new_port():
   return random.randint(10000, 65535)
 
 class TestGlobalScheduler(unittest.TestCase):
 
   def setUp(self):
-    # Start a Redis server.
+    # Start one Redis server and N pairs of (plasma, photon)
     redis_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), "../../core/src/common/thirdparty/redis/src/redis-server")
     redis_module = os.path.join(os.path.dirname(os.path.abspath(__file__)), "../../core/src/common/redis_module/libray_redis_module.so")
     assert os.path.isfile(redis_path)
     assert os.path.isfile(redis_module)
     node_ip_address = "127.0.0.1"
+    #redis_port = 6379 #default Redis port
     redis_port = new_port()
     redis_address = "{}:{}".format(node_ip_address, redis_port)
     self.redis_process = subprocess.Popen([redis_path, "--port", str(redis_port), "--loglevel", "warning", "--loadmodule", redis_module])
     time.sleep(0.1)
     # Create a Redis client.
     self.redis_client = redis.StrictRedis(host=node_ip_address, port=redis_port)
-    # Start the global scheduler.
+    # Start one global scheduler.
     self.p1 = global_scheduler.start_global_scheduler(redis_address, use_valgrind=USE_VALGRIND)
-    # Start the Plasma store.
-    plasma_store_name, self.p2 = plasma.start_plasma_store()
-    # Start the Plasma manager.
-    plasma_manager_name, self.p3, plasma_manager_port = plasma.start_plasma_manager(plasma_store_name, redis_address)
-    self.plasma_address = "{}:{}".format(node_ip_address, plasma_manager_port)
-    self.plasma_client = plasma.PlasmaClient(plasma_store_name, plasma_manager_name)
-    # Start the local scheduler.
-    local_scheduler_name, self.p4 = photon.start_local_scheduler(
-        plasma_store_name,
-        plasma_manager_name=plasma_manager_name,
-        plasma_address=self.plasma_address,
-        redis_address=redis_address)
-    # Connect to the scheduler.
-    self.photon_client = photon.PhotonClient(local_scheduler_name)
+    self.plasma_store_p2 = []
+    self.plasma_manager_p3 = []
+    self.local_scheduler_p4 = []
+    self.plasma_clients = []
+    self.photon_clients = []
+
+    for i in range(NUM_CLUSTER_NODES):
+      # Start the Plasma store. Plasma store name is randomly generated.
+      plasma_store_name, p2 = plasma.start_plasma_store()
+      self.plasma_store_p2.append(p2)
+      # Start the Plasma manager.
+      # Assumption: Plasma manager name and port are randomly generated by the plasma module.
```

Comment:
I had to check to make sure this is indeed the case, because we need different (name, port) pairs generated for the multinode test case. If the generated (name, port) pair were deterministic, I would've had to make changes inside the plasma module. I think explicitly stating this post-condition is useful.